.github/workflows/release.yml

```
@@ -13,93 +13,94 @@ on:
       - v[0-9]+.[0-9]+.[0-9]+
 
 env:
+  ZIG_VERSION: 0.13.0
+  GLIBC_VERSION: 2.17
   BIN_DIR: ${{ github.workspace }}/bin
-  MINIMUM_CMAKE_VERSION: '3.28'
 
-# Build on the oldest supported images, so we have broader compatibility
-# Build with gcc-10 to prevent triggering #14150 (default is still gcc-9 on 20.04)
+# Build with zig cc so we can target glibc 2.17, so we have broader compatibility
 jobs:
   linux:
-    runs-on: ubuntu-20.04
-    env:
-      CC: gcc-10
+    runs-on: ubuntu-latest
     outputs:
       version: ${{ steps.build.outputs.version }}
-    container:
-      image: ubuntu:18.04
-      options: --privileged # Privileged mode is needed to load fuse module.
     steps:
-      - name: Prepare container
+      - name: Install deps from apt
         run: |
-          apt-get update
-          apt-get install -y software-properties-common
-          add-apt-repository -y ppa:ubuntu-toolchain-r/test # For gcc-10.
-          add-apt-repository -y ppa:git-core/ppa # For git>=2.18.
-          apt-get update
-          apt-get install -y git gcc-10
-          apt-get install -y fuse libfuse2 # For linuxdeploy.
+          sudo apt-get update
+          sudo apt-get install -y fuse libfuse2 # For linuxdeploy.
+          sudo apt-get install -y build-essential curl gettext ninja-build unzip cmake
+          sudo apt-get install -y xz-utils # To extract zig
           # Workaround for https://github.com/actions/checkout/issues/766.
           git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
       - if: github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name != 'nightly'
         uses: actions/checkout@v3
         with:
           repository: 'neovim/neovim'
           ref: ${{ github.event.inputs.tag_name }}
           fetch-depth: 0
+
       - if: github.event_name == 'schedule' || github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name == 'nightly')
         uses: actions/checkout@v3
         with:
           repository: 'neovim/neovim'
           fetch-depth: 0
-      - run: |
-         apt-get update
-         apt-get install -y build-essential curl gettext ninja-build unzip
+
+      # zig cc with -02 implicitly adds -DNDEBUG so remove this from the generator flags
+      # Note needed for 0.10.1+ : https://github.com/neovim/neovim/pull/29599
```

```suggestion
      # Not needed for 0.10.1+ : https://github.com/neovim/neovim/pull/29599
```
Or `TODO(lewis): remove after 0.10.1 release`